unit-tests/catch.h

```diff
@@ -9,8 +9,15 @@
 // Otherwise, don't let it define its own:
 #define ELPP_NO_CHECK_MACROS
 
+// We always set our own main, CATCH_CONFIG_MAIN will cause errors in build due to having our main and catch2 main
+#if defined(CATCH_CONFIG_MAIN)
+#error CATCH_CONFIG_MAIN defined
+#endif
+
 #include "catch/catch.hpp"
+#include <string>
 
+std::string test_context;
```

Comment:
This should be in the .cpp, and referenced in the header as `extern` (the reverse of how it is now)

Reply:
Also, it shouldn't be in this header -- it should be in test.h, no?

Reply:
There is no test.h, do you want to add one and include in instead of catch.h in all places? seems silly

Reply:
Also I tries for a while and couldn't find how to use extern for a variable in a namespace...
`extern string test::context` gives an error of `name followed by '::' must be a class or namespace name`

Reply:
That's because `test` isn't a known namespace.
And there is a `test.h` -- you modified it :)

Reply:
Ok, so is there a way to make it known? 
If I put it in `test.h` how will the tests know about it? since they don't include `test.h`. Or do you want to say that if you want to use context you must include it?

Reply:
I'm saying it, yes: if someone needs to know what context they're in, they need to include `test.h`

Reply:
OK